grpclog/internal/loggerv2.go

```diff
@@ -108,69 +108,88 @@ type loggerT struct {
 	jsonFormat bool
 }
 
-func (g *loggerT) output(severity int, s string) {
+func (g *loggerT) printFunc(severity int, fn func() string) {
```

Comment:
nit: probably need to be more explicit here with func name or docstring that it doesn't attempt to output the `fn` string in case of discard

Reply:
I added a comment.